spacy/cli/debug_data.py

```diff
@@ -699,8 +699,33 @@ def _get_examples_without_label(data: Sequence[Example], label: str) -> int:
     return count
 
 
-def _get_labels_from_model(nlp: Language, pipe_name: str) -> Set[str]:
-    if pipe_name not in nlp.pipe_names:
-        return set()
-    pipe = nlp.get_pipe(pipe_name)
-    return set(pipe.labels)
+def _get_labels_from_model(
+    nlp: Language, factory_name: str
+) -> Set[str]:
+    pipe_names = [
+        pipe_name
+        for pipe_name in nlp.pipe_names
+        if nlp.get_pipe_meta(pipe_name).factory == factory_name
+    ]
+    labels: Set[str] = set()
+    for pipe_name in pipe_names:
+        pipe = nlp.get_pipe(pipe_name)
+        labels.update(pipe.labels)
+    return labels
+
+
+def _get_labels_from_spancat(
+    nlp: Language, factory_name: str
+) -> Dict[str, Set[str]]:
+    pipe_names = [
+        pipe_name
+        for pipe_name in nlp.pipe_names
+        if nlp.get_pipe_meta(pipe_name).factory == factory_name
+    ]
+    labels: Dict[str, Set[str]] = {}
+    for pipe_name in pipe_names:
+        pipe = cast(SpanCategorizer, nlp.get_pipe(pipe_name))
```

Comment:
I would personally do this as an `assert`. If this depends on user input/settings I would even do an `if` check and raise a `ValueError`, but I don't think that will be the case? 

```suggestion
        pipe = nlp.get_pipe(pipe_name)
        assert isinstance(pipe, SpanCategorizer)
```

Reply:
If mypy is fine with that, it's fine with me. Actually, there's no need for the `factory_name` arg, either...

Reply:
Yep, `mypy` parses the `assert` and will take that into account for the following lines :-)

Reply:
You could also have the `factory_name` set default to `"spancat"`?